Automattic/wp-calypso
# Sensei: Install additional plugins in Launch step

#### Proposed Changes

* Save selection from the Purpose screen in a session
* Install plugins selected there after the atomic site is ready in the Launch screen
* For WC and WC Payments, install the `woo-on-plan` software set

* Change the plugin slug used from `woothemes-sensei` to `sensei-pro`
* Also update the associated products in the plan selection screen
* Break up the plan setup component to more files & functions
* Make sure the page title is set (set this to 'Course Creator')
* Clear previous onboarding data in the first step (this fixes the domain search being pre-filled from a previous setup flow)

#### Testing Instructions

* Open the purpose site with a new or existing site at `calypso.localhost:3000/setup/sensei/senseiPurpose?siteSlug= URL`
* Select "Provide certification"
* Finish the setup flow
* Check that the 'Sensei LMS Certificates' plugin is installed
--
* ~For WooCommerce — this is currently activated anyway, because the Sensei Pro (woothemes-sensei) we add to cart requires it?~ Switched to the new `sensei-pro` plugin.
* To test the PR:
* Create a new site with a Business plan
* Check that the WooCommerce plugin is not installed. (To really check in the plugin screen, install an unrelated plugin to trigger an atomic transfer. Also, WooCommerce is shown in the menu even when the plugin is not active, as an upsell/starting point to set it up).
* Open this business site in the Sensei setup flow, by placing the url in `calypso.localhost:3000/setup/sensei/senseiPurpose?siteSlug= URL`
* Select "Sell courses and generate income"
* Finish the setup flow
* Check that the 'WooCommerce', 'WooCommerce Payments' and 'WooCommerce Google Analytics' plugins are installed and active

#### Pre-merge Checklist





#### Screenshots

<img width="1512" alt="image" src="https://user-images.githubusercontent.com/176949/217868974-619b5c30-a05c-4cb5-80a4-1e31b2848949.png">
